oscilloscope/agilent.py

```diff
@@ -0,0 +1,195 @@
+"""
+Wrappers for Agilent oscilloscopes.
+
+"""
+
+
+from __future__ import absolute_import
+import oscilloscope.wrappers as wrappers
+import labrad.units as U
+import struct
+import numpy as np
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+
+class AgilentDSOX4104AWrapper(wrappers.OscilloscopeWrapper):
+    """Wrapper for the Agilent DSO-X 4104A
+
+    We mostly override VISA string functions from the parent class, but in some
+    cases we have to write complete self-contained methods. This file should be
+    viewed along side the parent class in order to understand what's going on.
+
+    See programmer manual at:
+        http://www.keysight.com/upload/cmc_upload/All/4000_series_prog_guide.pdf?&cc=US&lc=eng
+    """
+
+    # CHANNEL
+
```

Comment:
LGTM.  

Note, the "...you're trading..." was a general 'you' and not directed at you personally.  =D